### ISSUE TITLE
Github comments not compliant w/ GFM soft line breaks?

### ISSUE THREAD

**Issue by AlecBenzer** [isaacs/github#1013]
Per GFM's spec: https://github.github.com/gfm/#soft-line-breaks

I would expect

```markdown
hello
world
```

to produce "hello world" on one line without a line break.

But in fact:

hello
world

This shows up as two separate lines.

CommonMark has the same in their spec, along with demo links: http://spec.commonmark.org/0.28/#soft-line-breaks

**Comment by AlecBenzer** [comment]
Response from Francis (GitHub Staff):

> Hi there Alec,
>
> Thanks for getting in touch! We've asked the team for their input on this one. We'll be sure to write back when we hear more from them.
>
> All the best,
> Francis

**Comment by NobbZ** [comment]
There seems to be some different behaviour in Comments or rendered documents.

In a comment in fact thats on two lines, but in comments a newline is always treated literally, even the following snippet will render on two distinct lines in a comment.

```md
hello
world
```

When rendered as a file as in my [experimental gist](https://gist.github.com/NobbZ/a605f645f9240b041b95dc488a284fb7) though, its on a single line.

Also, both versions are fine according to your link:

> A conforming parser may render a soft line break in HTML either as a line break or as a space.

**Comment by AlecBenzer** [comment]
@NobbZ 

> There seems to be some different behaviour in Comments or rendered documents.

Yeah, the answer here might just be that GitHub comments are not intended to use GFM. But previous discussions seemed to indicate that GFM should in fact be in use for comments (e.g., https://stackoverflow.com/a/21272554/598940).

> even the following snippet will render on two distinct lines in a comment

That's exactly the same snippet as in my original post, right?

**Comment by NobbZ** [comment]
Nope, my snippet doesn't have a trailing whitespace

**Comment by AlecBenzer** [comment]
Mine doesn't either. Are you looking at the highlighting to infer it does? On Sierra at least, that's just how newlines appear highlighted. E.g.: 
<img width="789" alt="screen shot 2017-09-09 at 12 15 20 pm" src="https://user-images.githubusercontent.com/92369/30241812-ae081940-9558-11e7-9cf7-281852ee6a0e.png">

**Comment by AlecBenzer** [comment]
Response from Francis (GitHub Staff):

> Hey Alec,
> 
> Thanks for hanging in there! Our team got back to us and I wanted to share their note with you.
> 
> In https://github.github.com/gfm/#soft-line-breaks, it says:
> 
> > A renderer may also provide an option to render soft line breaks as hard line breaks.
> 
> We used to render newlines as breaks in comments, so we use that option in comments, issues, and pull requests to provide consistency with past behavior. We don't currently use this for rendering blobs.
> 
> I hope that helps explain things, but let us know if you have any other questions!

That seems a little weird -- it's not really an option, it's just the way things work by default. It also feels weird for the standard to say something but then also just say it can be ignored. But I guess the idea is that this is just a weaker part of the standard.

**Comment by zevdg** [comment]
FYI, this seems to be a duplicate of https://github.com/github/markup/issues/1050 and a github engineer explained what was going on.  The issue there is about Github Flavored Markdown, but the answer works for the [commonmark spec](http://spec.commonmark.org/0.28/#soft-line-breaks) as well.

**Comment by jult** [comment]
Why does github ignore linebreaks when posting code in comments? This is just awfully weird and unfriendly.
`test
test`

should be posted on two lines.

**Comment by alecbz** [comment]
@jult Single backticks are specifically used for inline code, so that makes sense. If you use triple backticks newlines should show up.
